pulsar-broker/src/main/java/org/apache/pulsar/broker/PulsarService.java

```diff
@@ -162,7 +165,8 @@
     private ScheduledExecutorService compactorExecutor;
     private OrderedScheduler offloaderScheduler;
     private Offloaders offloaderManager = new Offloaders();
-    private LedgerOffloader offloader;
+    private LedgerOffloader defaultOffloader;
+    private Map<OffloadPolicies, LedgerOffloader> ledgerOffloaderMap = new ConcurrentHashMap<>();
```

Comment:
I'm a little worried about uses OffloadPolicies as a key, because is users frequently modified the policy, there will be some invalid Offloader objects here.